core/arch/x86/encode.c

```diff
@@ -1788,9 +1825,11 @@ encode_operand(decode_info_t *di, int optype, opnd_size_t opsize, opnd_t opnd)
     case TYPE_E:
     case TYPE_Q:
     case TYPE_W:
-    case TYPE_R:       /* we already ensured this is a reg, not memory */
-    case TYPE_P_MODRM: /* we already ensured this is a reg, not memory */
-    case TYPE_V_MODRM: /* we already ensured this is a reg, not memory */
+    case TYPE_K_MODRM:
```

Comment:
nit: Just seems an unusual ordering, K inserted up here, but K R appended?

Reply:
Ah ok yes I did that intentionally. The reason is that I put the "we already ensured this is a reg, not memory" one in a group. It's important to know that these ones are always (verified already) a register. The TYPE_K_MODRM can also be memory, so I pulled it up before the group starts. If that doesn't make sense please re-open.